collect_app/src/main/java/org/odk/collect/android/formmanagement/ServerFormsDetailsFetcher.java

```diff
@@ -159,16 +146,12 @@ private static boolean isMediaFileAlreadyDownloaded(List<File> localMediaFiles,
         return false;
     }
 
-    private String getMd5HashWithoutPrefix(String hash) {
-        return hash == null || hash.isEmpty() ? null : hash.substring("md5:".length());
+    private Form getFormByHash(String hashWithPrefix) {
+        String hash = hashWithPrefix.substring("md5:".length());
+        return formsRepository.getOneByMd5Hash(hash);
     }
 
-    private Form getFormByVersion(List<Form> forms, @Nullable String expectedVersion) {
-        for (Form form : forms) {
-            if (Objects.equals(form.getVersion(), expectedVersion)) {
-                return form;
-            }
-        }
-        return null;
+    private boolean isHashValid(String hash) {
+        return hash != null && hash.startsWith("md5:");
```

Comment:
I think we need a test for the invalid (doesn't start with `md5:` case)

Reply:
Like this https://github.com/getodk/collect/pull/4906/commits/ec965920fe9b15e78dee2867513851ec17f87d96?